src/ripple/protocol/impl/Rules.cpp

```diff
@@ -40,9 +39,8 @@ class Rules::Impl
         std::unordered_set<uint256, beast::uhash<>> const& presets,
         std::optional<uint256> const& digest,
         STVector256 const& amendments)
-        : presets_(presets)
+        : presets_(presets), digest_(digest)
```

Comment:
You need to swap the positions of these two member variables.  The `Rules::Impl` member variables are declared...
```
private:
    std::unordered_set<uint256, hardened_hash<>> set_;
    std::optional<uint256> digest_;
    std::unordered_set<uint256, beast::uhash<>> const& presets_;
```